### PR TITLE
Add -e back to core/database/test/scripts

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
# PR Description

# Tasks

* [ ] - A description of the PR has been provided, and a diagram included if it is a new feature.
* [ ] - Formatter has been run
* [ ] - CHANGELOG comment has been added
* [x] - Labels have been assigned to the pr
* [ ] - A reviwer has been added
* [x] - A user has been assigned to work on the pr
* [ ] - If new feature a unit test has been added

## Summary by Sourcery

Enhancements:
- Reintroduce the '-e' option in test scripts to ensure CI jobs fail when there are script errors, with a plan to handle residual password files through a separate cleanup script.

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
I approve once you get 1173 merged into this and the CI passes here.

**Review by AronPerez — APPROVED** [review]

File: .gitlab/end_to_end.yml
Comment (t-ramz) [reply .gitlab/end_to_end.yml]: Done

**Pull request merged**
